### PR TITLE
cURL extension for URLRequest

### PR DESCRIPTION
**Pull request by DimaZava**
### Added

- **URLRequest**
  - Added `curlString` property to get a cURL command representation of this URL request.

**NOTE:**  I didn't add tests because possible usage scenario is to export obtained cURL string to external app like terminal, Postman, Paw, etc. Please, feel free to comment and propose possible solutions of testing this extension.  

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

**Comment by omaralbeik** [comment]
Thanks for bringing this @DimaZava , I think you can still get the URL string for multiple success and failure scenarios, and compare them with what you expect from the extension

File: Sources/SwifterSwift/Foundation/URLRequestExtensions.swift [hunk 1]
```diff
@@ -20,6 +20,34 @@ public extension URLRequest {
         self.init(url: url)
     }
 
+    /// cURL command representation of this URL request.
```
Comment (mustafagunes) [inline Sources/SwifterSwift/Foundation/URLRequestExtensions.swift]: SwiftLint is warning. You should replace the comment line with the following example :)

`SwifterSwift: cURL command representation of this URL request.`

**Comment by LucianoPAlmeida** [comment]
@DimaZava Thanks for taking care of this one ;) 
Can you please add unit tests for this?

**Comment by DimaZava** [comment]
@omaralbeik Do you mean I can compare correct curl request text with one I obtain using extension? If yes - can you tell me where can I hold this text to compare? Create json, array, or?

**Comment by omaralbeik** [comment]
> @omaralbeik Do you mean I can compare correct curl request text with one I obtain using extension? If yes - can you tell me where can I hold this text to compare? Create json, array, or?

Exactly!
I don't think we need a json file for this, you can create a new test function in [URLExtensionsTests](https://github.com/SwifterSwift/SwifterSwift/blob/master/Tests/FoundationTests/URLExtensionsTests.swift) called `testCurlString` which has multiple scenarios for different urls and different requests and compare them with what you expect

File: Tests/FoundationTests/URLRequestExtensionsTests.swift [hunk 2]
```diff
@@ -14,6 +14,9 @@ import Foundation
 
 final class URLRequestExtensionsTests: XCTestCase {
 
+    private let insightNASAcURL = "curl https://api.nasa.gov/insight_weather/?api_key=mxbd8VDIy5CheCbrYgknXVH6X9ElpQaHMhne2YXP&feedtype=json&ver=1.0"
```
Comment (LucianoPAlmeida) [inline Tests/FoundationTests/URLRequestExtensionsTests.swift]: Can we move decl to local `testCUrlSring` in the function?

Comment (DimaZava) [reply Tests/FoundationTests/URLRequestExtensionsTests.swift]: Yep, don't even know why I decide to do like that

**Comment by SD10** [comment]
Thank you for contributing to SwifterSwift! I've invited you to join the SwifterSwift GitHub organization - no pressure to accept! If you'd like more information on what that means, check out our [contributing guidelines](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md). Feel free to reach out if you have any questions! 😃

**Comment by DimaZava** [comment]
@SD10 Wow, it's great honour for me! Of course I'd like to get more info on what that means 😃

**Comment by LucianoPAlmeida** [comment]
@DimaZava 
It means you are now part of the organization and part of the SwifterSwift contributors \o/ 
See more: https://github.com/orgs/SwifterSwift/teams/contributors
You also, now have access to the repo and can create branches directly on it instead of working from a fork :)

**Comment by DimaZava** [comment]
@LucianoPAlmeida Sounds awesome! Thanks!

**Pull request merged**